crates/shared/src/sources/balancer_v2/pool_init.rs

```diff
@@ -22,6 +22,7 @@ pub trait PoolInitializing: Send + Sync {
 ///
 /// This can be used to index all pools from events instead of relying on the
 /// Balancer subgraph for example.
+#[allow(dead_code)]
```

Comment:
They get better and better detecting dead code! :)

Reply:
I am not sure about plans for this struct, though. Maybe it can be removed.

Reply:
Let's remove dead code if we find it (or condition it on the right feature/cfg that uses it). Chances are it was just forgotten.